servers/rendering/renderer_rd/storage_rd/render_scene_buffers_rd.h

```diff
@@ -0,0 +1,273 @@
+	RID get_texture_slice(const StringName &p_context, const StringName &p_texture_name, const uint32_t p_layer, const uint32_t p_mipmap);
+
+	void clear_context(const StringName &p_context);
+
+	// Data buffers
+	bool has_data_buffer(const StringName &p_name);
+	void set_data_buffer(const StringName &p_name, Ref<RenderSceneBufferDataRD> p_buffer);
+	Ref<RenderSceneBufferDataRD> get_data_buffer(const StringName &p_name) const;
+
+	// Getters
+
+	_FORCE_INLINE_ RID get_render_target() const { return render_target; }
+	_FORCE_INLINE_ uint32_t get_view_count() const { return view_count; }
+	_FORCE_INLINE_ Size2i get_internal_size() const { return internal_size; }
+	_FORCE_INLINE_ Size2i get_target_size() const { return target_size; }
+	_FORCE_INLINE_ float get_fsr_sharpness() const { return fsr_sharpness; }
+	_FORCE_INLINE_ RS::ViewportMSAA get_msaa() const { return msaa; }
+	_FORCE_INLINE_ RS::ViewportScreenSpaceAA get_screen_space_aa() const { return screen_space_aa; }
+	_FORCE_INLINE_ bool get_use_taa() const { return use_taa; }
+	_FORCE_INLINE_ bool get_use_debanding() const { return use_debanding; }
+
+	uint64_t get_auto_exposure_version() const { return auto_exposure_version; }
+	void set_auto_exposure_version(const uint64_t p_auto_exposure_version) { auto_exposure_version = p_auto_exposure_version; }
+
+	// For our internal textures we provide some easy access methods.
+
+	_FORCE_INLINE_ RID get_internal_texture() const {
+		return get_texture(SNAME("RenderSceneBuffersRD"), SNAME("internal_texture"));
+	}
+	_FORCE_INLINE_ RID get_internal_texture(const uint32_t p_layer) {
+		return get_texture_slice(SNAME("RenderSceneBuffersRD"), SNAME("internal_texture"), p_layer, 0);
+	}
+
+	_FORCE_INLINE_ RID get_depth_texture() const {
+		return get_texture(SNAME("RenderSceneBuffersRD"), SNAME("depth_texture"));
+	}
+	_FORCE_INLINE_ RID get_depth_texture(const uint32_t p_layer) {
+		return get_texture_slice(SNAME("RenderSceneBuffersRD"), SNAME("depth_texture"), p_layer, 0);
+	}
+
+	// back buffer (color + depth)
+	RID get_back_buffer_texture() const { return blur[0].texture; } // We (re)use our blur texture here.
+
+	RID create_back_depth_texture();
+	_FORCE_INLINE_ RID get_back_depth_texture() { return has_texture(SNAME("RenderSceneBuffersRD"), SNAME("back_depth_texture")) ? get_texture(SNAME("RenderSceneBuffersRD"), SNAME("back_depth_texture")) : RID(); }
+	_FORCE_INLINE_ RID get_back_depth_texture(const uint32_t p_layer) { return has_texture(SNAME("RenderSceneBuffersRD"), SNAME("back_depth_texture")) ? get_texture_slice(SNAME("RenderSceneBuffersRD"), SNAME("back_depth_texture"), p_layer, 0) : RID(); }
+
+	// Velocity, currently only used by TAA (Clustered) but we'll be using this in other places soon too.
+
+	void ensure_velocity();
+	bool has_velocity_buffer(bool p_has_msaa) { return has_texture(SNAME("RenderSceneBuffersRD"), p_has_msaa ? SNAME("msaa_velocity") : SNAME("velocity")); }
```

Comment:
I think for the categories and commonly used textures, it should be fine to have a .h that does defines like:

```c++
#define RB_SCOPE_BUFFERS SNAME("render_buffers")
#define RB_SCOPE_VRS SNAME("VRS")
// and
#define RB_TEX_VELOCITY SNAME("velocity")
#define RB_TEX_COLOR SNAME("color")
#define RB_TEX_DEPTH SNAME("depth")

// so you can do like:
rb->get_texture(RB_SCOPE_BUFFERS,RB_TEX_COLOR);
rb->get_texture(RB_SCOPE_BUFFERS,RB_TEX_DEPTH);
```
Just for the sake of not using strings if not needed. They can be defined locally also in cpp files for custom effects and the likes, so no need to make all global.

Reply:
Ah we discussed this on chat and I actually moved away from this after our discussion. I agree, having defines makes it much less likely to make mistakes.